packages/core/src/sync-historical/index.ts

```diff
@@ -291,103 +297,84 @@ export const createHistoricalSync = async (
     await Promise.all(requiredBlocks.map((b) => syncBlock(BigInt(b))));
   };
 
-  const syncTraceFilter = async (
-    filter: CallTraceFilter,
-    interval: Interval,
+  const syncTrace = async (
+    filter: TransactionFilter | TransferFilter,
+    blockNumber: number,
   ) => {
-    // Resolve `filter.toAddress`
-    let toAddress: Address[] | undefined;
-    if (isAddressFactory(filter.toAddress)) {
-      const childAddresses = await syncAddress(filter.toAddress, interval);
-      if (
-        childAddresses.length < args.common.options.factoryAddressCountThreshold
-      ) {
-        toAddress = childAddresses;
-      } else {
-        toAddress = undefined;
-      }
-    } else {
-      toAddress = filter.toAddress;
-    }
-
-    if (isKilled) return;
-
-    let callTraces = await _trace_filter(args.requestQueue, {
-      fromAddress: filter.fromAddress,
-      toAddress,
-      fromBlock: interval[0],
-      toBlock: interval[1],
-    }).then(
-      (traces) =>
-        traces.flat().filter((t) => t.type === "call") as SyncCallTrace[],
-    );
+    const syncTraces = await _debug_traceBlockByNumber(args.requestQueue, {
+      blockNumber: toHex(blockNumber),
+    });
```

Comment:
We should start to think about how we can avoid requesting the same block twice if there are multiple transfer and transaction filters